src/error.rs

```diff
@@ -18,8 +18,6 @@ pub enum ProofError {
 pub enum LurkError {
     #[error("Evaluation error: {0}")]
     Eval(String),
-    #[error("Reduction error: {0}")]
-    Reduce(String),
```

Comment:
If we're collapsing these, it probably makes more sense to retain `Reduce` (even though I understand the initial error-handling adventitiously assigned most reduction errors to `Eval`).